qutebrowser/completion/models/miscmodels.py

```diff
@@ -112,6 +112,9 @@ def delete_buffer(data):
 
     model = completionmodel.CompletionModel(column_widths=(6, 40, 54))
 
+    tabs_are_windows = config.val.tabs.tabs_are_windows
+    windows = []  # list storing single-tab windows when tabs_are_windows
```

Comment:
This [needs a type annotation](https://travis-ci.org/qutebrowser/qutebrowser/jobs/626559628#L258) - probably something like `# type: typing.List[typing.Tuple[str, str, str]]` if I'm not mistaken.

Reply:
Alright, I'll be sure to add that in. I saw that error in Travis, but wasn't sure whether I should add it, since similarly used variables like `tabs` inside the for-loop few lines done didn't have it.

In what situations are type annotations actually required to pass MyPy checks?

Reply:
It's required as soon as [mypy can't infer it](https://mypy.readthedocs.io/en/latest/type_inference_and_annotations.html#explicit-types-for-collections) - but nwo that you mention it, I'm not sure why it doesn't complain about the type for `tabs` below! I tried adding a `reveal_type(tabs)` for mypy, but as soon as I do that, I *do* get `error: Need type annotation for
'tabs'`. Could you add it there as well?